contracts/FungibleTokenMetadataViews.cdc

```diff
@@ -0,0 +1,175 @@
+import FungibleToken from "./FungibleToken.cdc"
+import MetadataViews from "./utilityContracts/MetadataViews.cdc"
+
+/// This contract implements the metadata standard proposed
+/// in FLIP-1087.
+/// 
+/// Ref: https://github.com/onflow/flow/blob/master/flips/20220811-fungible-tokens-metadata.md
+/// 
+/// Structs and resources can implement one or more
+/// metadata types, called views. Each view type represents
+/// a different kind of metadata.
+///
+pub contract FungibleTokenMetadataViews {
+    /// FTView wraps FTDisplay and FTVaultData, and is used to give a complete 
+    /// picture of a Fungible Token. Most Fungible Token contracts should 
+    /// implement this view.
+    ///
+    pub struct FTView {
+        pub let ftDisplay: FTDisplay?     
+        pub let vaultData: FTVaultData?
+        init(
+            ftDisplay: FTDisplay?,
+            vaultData: FTVaultData?
+        ) {
+            self.ftDisplay = ftDisplay
+            self.vaultData = vaultData
+        }
+    }
+
+    /// Helper to get a FT view 
+    ///
+    /// @param viewResolver: A reference to the resolver resource
+    /// @return A FTView struct
+    ///
+    pub fun getFTView(viewResolver: &{MetadataViews.Resolver}): FTView {
+        let ftView = viewResolver.resolveView(Type<FTView>())
+        if ftView != nil {
+            return ftView! as! FTView
+        }
+        return FTView(
+            ftDisplay: self.getFTDisplay(viewResolver),
+            vaultData : self.getFTVaultData(viewResolver)
+        )
+    }
+
+    /// View to expose the information needed to showcase this FT. 
+    /// This can be used by applications to give an overview and 
+    /// graphics of the FT.
+    ///
+    pub struct FTDisplay {
+        /// Name that should be used when displaying this FT.
+        pub let name: String
+
+        /// Symbol that could be used as a shorter name for the FT.
+        pub let symbol: String
+
+        /// Description that should be used to give an overview of this FT.
+        pub let description: String
```

Comment:
It'd be nice to include examples for all the fields. For example:

```suggestion
        /// The display name for this token.
        ///
        /// Example: "Flow"
        ///
        pub let name: String

        /// The abbreviated symbol for this token.
        ///
        /// Example: "FLOW"
        ///
        pub let symbol: String

        /// A description the provides an overview of this token.
        ///
        /// Example: "The FLOW token is the native currency of the Flow network."
        ///
        pub let description: String
```

Reply:
Yeah that makes total sense